packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4218,6 +4218,14 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     return null;
   }
 
+  /// A cache of whether this [Element] depends on an [InheritedElement] with
+  /// [InheritedElement.clearDependencyOnRebuild] as true.
+  ///
+  /// This avoids having to iterate over the list of all dependencies whenever
+  /// the widget rebuilds.
+  /// This only works if we assume that [InheritedElement.clearDependencyOnRebuild]
+  /// never changes.
+  bool _hasDependencyWhichNeedsClearingOnRebuild = false;
```

Comment:
I am actually wondering: Would it simplify things if we just track the dependencies that we need to clear in a separate list instead of having this flag?

Reply:
That would be a lot less efficient though, wouldn't it?

Reply:
Can you elaborate?

Reply:
It's slower to push into a list than to do a simple boolean operation.

And I'll admit, I don't see why a list  would simplify the logic.